flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElectionServiceTest.java

```diff
@@ -210,6 +212,33 @@ public void testErrorIsIgnoredAfterBeingStop() throws Exception {
 		}};
 	}
 
+	/**
+	 * Tests that we can shut down the DefaultLeaderElectionService if the used
+	 * LeaderElectionDriver holds an internal lock. See FLINK-20008 for more details.
+	 */
+	@Test
```

Comment:
I think this test could also pass in most cases without moving `leaderElectionDriver.close()` out of `lock`. But it will fail in some corner situation. Right?

Reply:
yes exactly